Src/MoneyFox.Windows.Cortana/CortanaCommands.cs

```diff
@@ -0,0 +1,668 @@
+﻿using EntityFramework.DbContextScope;
+using MoneyFox.DataAccess.DataServices;
+using MoneyFox.DataAccess.Pocos;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Windows.ApplicationModel.AppService;
+using Windows.ApplicationModel.Background;
+using Windows.ApplicationModel.Resources.Core;
+using Windows.ApplicationModel.VoiceCommands;
+using Windows.Storage;
+using Windows.System.Threading;
+using System.Linq;
+using Newtonsoft.Json;
+using Windows.ApplicationModel.Resources;
+using System.Threading.Tasks;
+using MoneyFox.Foundation;
+using MoneyFox.DataAccess.Entities;
+using MoneyFox.Foundation.Resources;
+using stor = Windows.Storage;
+using System.Threading;
+
+namespace MoneyFox.Windows.Cortana
+{
+    public sealed class CortanaCommands : IBackgroundTask
+    {
+        BackgroundTaskDeferral serviceDeferral;
+        PaymentEntity payment;
+        AccountEntity account;
+        AccountService Accounts;
+        PaymentService Payments;
+      
+
+        public async void Run(IBackgroundTaskInstance taskInstance)
+        {
+            serviceDeferral = taskInstance.GetDeferral();
+            taskInstance.Canceled += TaskInstance_Canceled;
+            var local_folder = ApplicationData.Current.LocalFolder;
+            AppServiceTriggerDetails trigger = taskInstance.TriggerDetails as AppServiceTriggerDetails;
+            VoiceCommandServiceConnection connection = VoiceCommandServiceConnection.FromAppServiceTriggerDetails(trigger);
+            VoiceCommand voiceCommand = await connection.GetVoiceCommandAsync();
+            VoiceCommandUserMessage userMessage;
+            VoiceCommandUserMessage repromptMessage;
+            var rule = voiceCommand.SpeechRecognitionResult.RulePath[0];
+            Accounts = new AccountService(new AmbientDbContextLocator(), new DbContextScopeFactory());
+            Payments = new PaymentService(new AmbientDbContextLocator(), new DbContextScopeFactory());
+            string commandType = "";
+
+            switch (rule)
+            {
+                case "create-payment":
```

Comment:
Is it possible to extract the logic in the cases into own classes and use for example a factory pattern? such large methods / switch cases tend to get hard to maintain.

Reply:
i believe there might be a way to extract it to their own background task

Reply:
i created each task its own app service, not optimal, i might be able to create a strategy pattern but not seeing a way to create a factory pattern.